templates/_macros/hero.jinja

```diff
@@ -0,0 +1,127 @@
+  title_text='',
+  subtitle_text='',
+  layout="fallback",
+  is_split_on_tablet = false,
+  is_split_on_desktop = true,
+  is_image_full_cover = false,
+  cta_primary_url = '#',
+  cta_secondary_url = '#'
+) -%}
+  {% set has_subtitle = subtitle_text|trim|length > 0 %}
+  {% set description_content = caller('description') %}
+  {% set has_description = description_content|trim|length > 0 %}
+  {% set cta_primary_content = caller('cta_primary') %}
+  {% set cta_secondary_content = caller('cta_secondary') %}
+  {% set has_cta = cta_primary_content|trim|length > 0 %}
+  {% set has_cta_secondary = cta_secondary_content|trim|length > 0 %}
+  {% set image_content = caller('image') %}
+
+  {# User can pass layout as "X-Y" or "X/Y" #}
+  {% set layout = layout | trim | replace('/', '-') %}
+  {% if layout not in ['50-50', '75-25', '25-75', 'fallback'] %}
+    {% set layout = 'fallback' %}
+  {% endif %}
+
+  {# Full cover image only supported on 50-50 layout #}
+  {% set is_image_full_cover = is_image_full_cover and layout == '50-50' %}
+
+  {# Construct grid classes from params input #}
+  {% set row_classes = [] %}
+  {% set col_classes = ['col', 'col'] %}
+  {% if layout in ['50-50', '25-75'] %}
+    {# Responsive breakpoint classes were requested #}
+    {% if is_split_on_tablet or is_split_on_desktop %}
+      {% if is_split_on_tablet %}
+        {{ row_classes.append('row--' + layout + '-on-medium') or "" }}
+      {% endif %}
+      {% if is_split_on_desktop %}
+        {{ row_classes.append('row--' + layout + '-on-large') or "" }}
+      {% endif %}
+    {% else %}
+      {{ row_classes.append('row--' + layout) or "" }}
+    {% endif %}
+  {% elif layout  == '75-25' %}
+    {# there is no .row--75-25 class, so we use unmodified row and col-9 and col-3 to create 75-25 layout #}
+    {{ row_classes.append('row') or "" }}
+    {% set col_classes = ['col-9', 'col-3'] %}
+  {% endif %}
+
+  {% macro _hero_title_block() -%}
+    {# On full-cover, the h1 and h2 are in separate columns, so there must be no margin-bottom to keep h1 close to h2 on smaller screens  #}
+    <h1 class="{% if is_image_full_cover %}u-no-margin--bottom{% endif %}">
+      {{ title_text }}
+    </h1>
+  {% endmacro %}
+
+  {% macro _hero_subtitle_block() -%}
+    {% if has_subtitle %}
+      <h2>
+        {{ subtitle_text }}
+      </h2>
+    {% endif %}
+  {% endmacro %}
+
+  {% macro _hero_cta_block() -%}
+    {% if has_cta %}
+      <div class="p-cta-block">
+        <a href="{{ cta_primary_url }}" class="p-button--positive">{{ cta_primary_content }}</a>
+        {% if has_cta_secondary -%}
+          <a href="{{ cta_secondary_url }}">{{ cta_secondary_content }}</a>
+        {%- endif %}
+      </div>
+    {% endif %}
+  {% endmacro %}
+
+  {% macro _hero_description_block() -%}
+    {% if has_description %}
+      <div class="p-section--shallow">
+        {{ description_content }}
+      </div>
+    {% endif %}
+  {% endmacro %}
+
+  <section class="p-section--hero">
```

Comment:
I know that potentially it would be a little bit of duplication, but maybe it would be clearer to have the `{% if is_image_full_cover %}` conditional on top level? To avoid checking for it 3 times inside? Currently it's quite hard to understand what the actual structure will end up being.

Suggestion:

```html
<section class="p-section--hero">
  <div class="{{ row_classes | join(' ') }}">
    {% if is_image_full_cover %}
        <!-- full width image structure -->
    {% else %}
        <!-- other  -->
    {% endif %}
  </div>
</section>
```

Reply:
Implemented